pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2628,6 +2628,37 @@ public void handleCommandStatus(String status, long updateCount, long insertOID)
     }
   }
 
+  @Override
+  public boolean requiresDescribe(Query query, ParameterList preparedParameters) {
+    if (preparedParameters.getParameterCount() == 0) {
+      return false;
+    }
+    // See org.postgresql.core.v3.QueryExecutorImpl.sendQuery
+    Query[] subqueries = query.getSubqueries();
+    SimpleParameterList[] subparams = ((V3ParameterList) preparedParameters).getSubparams();
+    if (subqueries == null) {
```

Comment:
This looks suspicious. Did you intend to check for `subqueries == null`?
If so, would you move the condition right after `subqueries` variable initialization?

Could `subparams` be `null`? Should we check it as well?

Reply:
> This looks suspicious. Did you intend to check for subqueries == null?

Exactly, thats what I'm doing here

> If so, would you move the condition right after subqueries variable initialization?

I agree, done

> Could subparams be null? Should we check it as well?

They cannot. The `subparams` are only present in case when we're dealing with `CompositeQuery` like this:

```

```

But If we're dealing with the composite query, the `subqueries` are guaranteed to be not `null`. In other words, it is by design that either `subqueries == null` AND  `subparams == null`, or they are both present. 

**However**, I think we can do better and just do an `query instanceOf CompositeQuery` here, to eliminate such question in the first place. Do you agree?